packages/tools/src/stateManagement/segmentation/polySeg/Labelmap/convertContourToLabelmap.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  StackViewport,
+  Types,
+  VolumeViewport,
+  cache,
+  utilities,
+  getWebWorkerManager,
+  volumeLoader,
+  imageLoader,
+  metaData,
+  Enums,
+  triggerEvent,
+  eventTarget,
+} from '@cornerstonejs/core';
+import {
+  Annotation,
+  ContourSegmentationData,
+  PolySegConversionOptions,
+} from '../../../../types';
+import { getAnnotation } from '../../..';
+import { vec3 } from 'gl-matrix';
+import { Events } from '../../../../enums';
+
+const workerManager = getWebWorkerManager();
+
+export async function convertContourToVolumeLabelmap(
+  contourRepresentationData: ContourSegmentationData,
+  options: PolySegConversionOptions = {}
+) {
+  const { viewport } = options;
+
+  let imageIds;
+  if (viewport instanceof VolumeViewport) {
+    const defaultActor = viewport.getDefaultActor();
+    const volumeId = defaultActor.uid;
+    const volume = cache.getVolume(volumeId);
+    imageIds = volume.imageIds;
+  } else if (viewport instanceof StackViewport) {
```

Comment:
How about  if( (viewport as IStackViewport).getImageIds ) ...
That way it works for anything that can get a list of imageIds.  Even better, for VolumeViewport, you could implement getImageIds as the code you just have above, then you don't need to care about whether it is volume or stack or video or ....